ui/advanced-widgets.md

```diff
@@ -0,0 +1,225 @@
+---
+title: Advanced Custom Widgets
+---
+
+# Custom Widgets - Advanced Topics
+
+[[toc]]
+
+This page expands of some of the topics initially introduced in [Building Pages](building-pages.html).
+
+## Advanced Components
+
+In the custom widget system there are many options for different components to include beyond the standard OH set.  Where each component comes from determines many of the options available for configuration and styling.
+
+### F7 components
+
+The MainUI is built primarily using the [Framework7](https://framework7.io/) library.  Because most of the OH specific components are built on top of these f7 components, [the documentation of the f7 components](https://framework7.io/vue/) is useful for users wishing to understand more about the OH components as well.  In addition, however, the f7 components themselves are available as options in the widget editor.
+
+As a general rule, the f7 components will have more configuration and style flexibility than the OH counterparts.  So, their use is recommended when there is something about the component that needs to be configured in a way different than what is set in the OH version.  Of course, the f7 components do not have the OH specific functions available, so while they can have values based on items using the epxression system, they cannot easily be used to trigger rules, update items or variables, etc.
+
+The most commonly used f7 components will likely be the `f7-block`, `f7-row`, and `f7-col`.  These all generate a simple `<div>` element with one base f7 class (`block`, `row`, and `col` respectively).  These components are therefore useful as fundamental building-blocks of widget or page. The list components `f7-list-item` and `f7-list-item-row` can often be useful as well given the flexibility they provide for complex structure inside a list.
+
+Any of the OH components that allow for [wiget actions](building-pages.html#actions), include easy configuration for using some other widget as a popup or popover.  If, however, there is need for the popup or popover to be built-in with a single widget (e.g., to add a widget to the [marketplace](https://community.openhab.org/c/marketplace/ui-widgets/75) that includes a popup or popover), the `f7-popup` and `f7-popover` component can be used and the open or closed status of that modal object controled by the `popupOpen`, `popupClose`, `popoverOpen`, and `popoverClose` properties available in many of the other f7 components and their OH derivaties.
```

Comment:
I am not sure if `Any of the OH components that allow widget actions` (without the `for`) is better.